app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java

```diff
@@ -0,0 +1,78 @@
+package ingsw.group1.findmyphone.contacts;
+
+import android.widget.Filter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Class extends {@link Filter}
+ * used to filter contacts by name and address.
+ * Research isn't case sensitive.
+ *
+ * @author Giorgia Bortoletti
+ */
+class ContactFilter<C extends SMSContact> extends Filter {
+
+    private List<C> allContacts;
+    private List<C> selectedContacts;
+    private SMSContactRecyclerAdapter adapter;
+
+    /**
+     * Constructor
+     *
+     * @param adapter          to notify contacts list change
```

Comment:
Try writing specifics starting with capital letters, like:
`@param adapter          `**T**`o notify contacts list change`
Someone from group 1 suggested me just that and it makes specifics easier to read

Reply:
Thanks, I'll modify all specs.